### DISCUSSION TITLE
ready for CMake 3.12 ?

### DISCUSSION THREAD

**Discussion by jwuttke** [RosettaCommons/binder#185] [General]
Is there still need to support CMake 2.8? Requiring a more recent version would allow to simplify things. For instance, with CMake 3.12 one could use out-of-the-box support for Python3, which would allow removal of cmake/Modules/FindPython*.

**Comment by jwuttke** [comment]
Same question should be asked regarding Python2, now officially dead.

**Reply by andriish** [reply]
Hi @jwuttke ,

- One can bump the minimal version of cmake to 3.0 or so. True.

- But the python modules in cmake directory are customized to support builds with multiple python versions simultaneously, e.g. `BINDER_TEST_PYTHON_VERSIONS=2.7,3.9,3.10`. This feature is not present in the upstream python cmake modules. I've tried to push it upstream but w/o much success. If you want we can try to push it once again.

- Python2 is still very widely used and it will remain so in the next 5 years, I think.

Best regards,

Andrii

**Comment by jwuttke** [comment]
Thanks, Andrii. Maybe we better concentrate on other things, and come back to this in five years. Cheers, Joachim
